### PR TITLE
PM-4960 Migrate Verify Recover Delete Component

### PR DESCRIPTION
**Pull request by KiruthigaManivannan**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [x] Other
```

## Objective

Migrate the Verify Recover Delete Component to use the component library.

## Code changes

- **verify-recover-delete.component.html:** Added form group and bit form fields
- **verify-recover-delete.component.ts:** Added new form group , converted submit to arrow function
- **oss-routing.module.ts:** Added Routing for Anon layout wrapper component.

## Screenshots

https://github.com/bitwarden/clients/assets/162679756/d8c52767-5dbf-420b-845b-a99d800d6edb

https://github.com/bitwarden/clients/assets/162679756/d8c52767-5dbf-420b-845b-a99d800d6edb

https://github.com/bitwarden/clients/assets/162679756/d8c52767-5dbf-420b-845b-a99d800d6edb

### CODE REVIEW

**Review by ike-kottlowski — APPROVED** [review]
functional tests were good.

File: apps/web/src/app/oss-routing.module.ts [hunk 1]
```diff
@@ -295,6 +290,20 @@ const routes: Routes = [
...
+  {
+    path: "",
+    component: AnonLayoutWrapperComponent,
+    children: [
+      {
+        path: "verify-recover-delete",
+        component: VerifyRecoverDeleteComponent,
+        canActivate: [UnauthGuard],
+        data: {
+          pageTitle: "deleteAccount",
```
Comment (rr-bw) [inline apps/web/src/app/oss-routing.module.ts:302]: @KiruthigaManivannan When you get the latest from `main` and resolve the merge conflict in oss-routing.module.ts, could you also carry over the `titleId`?

```typescript
data: {
  pageTitle: "deleteAccount",
  titleId: "deleteAccount",
}
```

You do _not_ need to add `satisfies AnonLayoutWrapperData` to the object. Auth team can do that later as we are still working on the type safety and don't want to block this PR.

File: apps/web/src/app/oss-routing.module.ts [hunk 2]
```diff
@@ -295,6 +290,20 @@ const routes: Routes = [
...
       import("./admin-console/organizations/organization.module").then((m) => m.OrganizationModule),
   },
+  {
+    path: "",
+    component: AnonLayoutWrapperComponent,
+    children: [
+      {
+        path: "verify-recover-delete",
+        component: VerifyRecoverDeleteComponent,
+        canActivate: [UnauthGuard],
```
Comment (ike-kottlowski) [inline apps/web/src/app/oss-routing.module.ts:300]: Is it possible to change this to the `UnauthGuardFn()`?

**Review by ike-kottlowski — CHANGES_REQUESTED** [review]

**Review by ike-kottlowski — DISMISSED** [review]
Looks good.

**Review by ike-kottlowski — APPROVED** [review]

**Review by ike-kottlowski — APPROVED** [review]

**Pull request merged**